### PR TITLE
Short verification_uri for Device Authorization Request

### LINKED ISSUE #16107: Short verification_uri for Device Authorization Request
### Description

Admin user should be able to configure optionally per realm a shorten url for showing in verification_uri. Redirection to Keycloak  verification URI ({server_url}/realms/{realmname}/device) would be done outside Keycloak - fe in a proxy. 

### Discussion

https://github.com/keycloak/keycloak/discussions/15004

### Motivation

[ Device Authorization Response documentation](https://www.rfc-editor.org/rfc/rfc8628.html#section-3.2) is written :

> verification_uri
      REQUIRED.  The end-user verification URI on the authorization
      server.  The URI should be short and easy to remember as end users
      will be asked to manually type it into their user agent.

Microsoft for example does something like this with https://microsoft.com/devicelogin forwards to https://login.microsoftonline.com/common/oauth2/deviceauth.

### Details

_No response_

### PR DESCRIPTION
**Pull request by cgeorgilakis**
Closes #16107

Related [PR to Keycloak admin-ui](https://github.com/keycloak/keycloak-ui/pull/4081)

### CODE REVIEW

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/OAuth2DeviceAuthorizationGrantTest.java [hunk 1]
```diff
@@ -75,6 +75,7 @@ public class OAuth2DeviceAuthorizationGrantTest extends AbstractKeycloakTest {
     public static final String DEVICE_APP = "test-device";
     public static final String DEVICE_APP_PUBLIC = "test-device-public";
     public static final String DEVICE_APP_PUBLIC_CUSTOM_CONSENT = "test-device-public-custom-consent";
+    private static final String shortDeficeFlowUrl = "https://keycloak.org/device";
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/OAuth2DeviceAuthorizationGrantTest.java:78]:
```suggestion
    public static final String SHORT_DEVICE_FLOW_URL = "https://keycloak.org/device";
```

File: services/src/main/java/org/keycloak/protocol/oidc/grants/device/endpoints/DeviceEndpoint.java [hunk 2]
```diff
@@ -80,6 +80,8 @@ public class DeviceEndpoint extends AuthorizationEndpointBase implements RealmRe
 
     protected static final Logger logger = Logger.getLogger(DeviceEndpoint.class);
 
+    private String SHORT_VERIFICATION_URI = "shortVerificationUri";
```
Comment (mposolda) [inline services/src/main/java/org/keycloak/protocol/oidc/grants/device/endpoints/DeviceEndpoint.java:83]:
```suggestion
    public String SHORT_VERIFICATION_URI = "shortVerificationUri";
```

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/OAuth2DeviceAuthorizationGrantTest.java [hunk 3]
```diff
@@ -217,6 +218,32 @@ public void testPublicClient() throws Exception {
...
     }
 
+
+    @Test
+    public void testCustomVerificationUri() throws Exception {
+        // Device Authorization Request from device
+        try {
+            RealmResource testRealm = adminClient.realm(REALM_NAME);
+            RealmRepresentation realmRep = testRealm.toRepresentation();
+            realmRep.getAttributes().put("shortVerificationUri", shortDeficeFlowUrl);
```
Comment (mposolda) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/OAuth2DeviceAuthorizationGrantTest.java:228]:
```suggestion
            realmRep.getAttributes().put(DeviceEndpoint.SHORT_VERIFICATION_URI, SHORT_DEVICE_FLOW_URL);
```

**Review by mposolda — CHANGES_REQUESTED** [review]
@cgeorgilakis Thanks, I've added few minor suggestions inline to use constants and align constant `shortDeviceFlowUrl` in the OAuth2DeviceAuthorizationGrantTest with other constants in that class.

I did not verified if my suggestions compiles as is (I guess rather not and some more minor changes might be needed...)

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/OAuth2DeviceAuthorizationGrantTest.java (hunk 1 above)
Comment (cgeorgilakis) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/OAuth2DeviceAuthorizationGrantTest.java:78]: Why make public a variable that is only used in this class?

Comment (mposolda) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/OAuth2DeviceAuthorizationGrantTest.java:78]: Yes, you're right. I was just wondering about making it consistent with other constants.

So when you're on this, could you leave it private, but possibly also make other variables in this class private (DEVICE_APP and others) as they are also used just in this test?

Will be good to fix typo (defice) and use the notation for constants like SHORT_DEVICE_FLOW_URL.

**Comment by cgeorgilakis** [comment]
I have made the proposed changes.

**Review by mposolda — APPROVED** [review]
@cgeorgilakis Thanks! I hope to merge once tests are OK

**Pull request merged**